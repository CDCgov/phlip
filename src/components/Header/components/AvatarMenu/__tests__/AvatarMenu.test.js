import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { AvatarMenu } from '../index'

const props = {
  role: 'Admin',
  initials: 'TU',
  open: false,
  menuAnchor: null,
  onLogoutUser: jest.fn(),
  onToggleMenu: jest.fn(),
  onCloseMenu: jest.fn()
}

const setup = otherProps => {
  return mount(
    <MemoryRouter>
      <AvatarMenu {...props} {...otherProps} />
    </MemoryRouter>
  )
}

describe('Header -- AvatarMenu', () => {
  test('should render correctly', () => {
    expect(shallow(<AvatarMenu {...props} />)).toMatchSnapshot()
  })

  test('should call to open menu when Avatar is clicked', () => {
    const wrapper = setup()
    wrapper.find('Avatar').at(0).simulate('click')
    wrapper.update()
    expect(wrapper.find('AvatarMenu').prop('onToggleMenu')).toHaveBeenCalled()
  })

  test('should hide `admin` option from menu if role is not admin', () => {
    const wrapper = setup({ role: 'Coordinator', open: true })
    expect(wrapper.find('MenuList').find('MenuItem').length).toEqual(1)
    expect(wrapper.find('MenuList').find('MenuItem').text()).toEqual('Logout')
  })

  test('should open the menu when open is true', () => {
    const wrapper = setup({ open: true })
    expect(wrapper.find('MenuList').exists()).toBeTruthy()
  })

  test('should call logout when the logout menu item is clicked', () => {
    const wrapper = setup({ open: true })
    wrapper.find('MenuList').find('MenuItem').at(0).simulate('click')
    wrapper.update()
    expect(wrapper.find('AvatarMenu').prop('onLogoutUser')).toHaveBeenCalled()
  })
})