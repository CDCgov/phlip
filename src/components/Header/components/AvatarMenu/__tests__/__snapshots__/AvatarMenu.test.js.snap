// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header -- AvatarMenu should render correctly 1`] = `
<ClickAwayListener
  onClickAway={[Function]}
>
  <WithStyles(GridWrapper)
    item={true}
    style={
      Object {
        "zIndex": 2,
      }
    }
  >
    <Manager
      tag="div"
    >
      <Target>
        <Avatar
          big={false}
          initials="TU"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />
      </Target>
    </Manager>
  </WithStyles(GridWrapper)>
</ClickAwayListener>
`;
